Mesh_3/include/CGAL/Mesh_3/Protect_edges_sizing_field.h

```diff
@@ -1502,9 +1538,42 @@ do_balls_intersect(const Vertex_handle& va, const Vertex_handle& vb) const
   return do_intersect(sphere(cp(wa), cw(wa)), sphere(cp(wb), cw(wb)));
 }
 
-template <typename C3T3, typename MD, typename Sf>
-typename Protect_edges_sizing_field<C3T3, MD, Sf>::Vertex_handle
-Protect_edges_sizing_field<C3T3, MD, Sf>::
+template <typename C3T3, typename MD, typename Sf, typename Df>
+bool
+Protect_edges_sizing_field<C3T3, MD, Sf, Df>::
+approx_is_too_large(const Vertex_handle& va, const Vertex_handle& vb) const
+{
+  if ( ! c3t3_.is_in_complex(va, vb) )
+  {
+    return false;
+  }
```

Comment:
This test is rather expensive.
Is it possible to avoid it, if `approx_is_too_large()` is called only on complex edges?

Then it could be turned into an assertion

Reply:
It would be possible. But the method `non_adjacent_but_intersect` called just before also has this check :
```
      if( non_adjacent_but_intersect(va, vb) <-- cannot be true for complex edges
          || (check_edge_distance && approx_is_too_large(va, vb))) <-- cannot be true for non-complex edges
```
Maybe these two checks could be separated in two loops, one with only non-complex edges and one with only in-complex edges.